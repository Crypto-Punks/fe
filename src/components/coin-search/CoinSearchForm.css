.SearchForm {
  position: fixed;
  width: 20%;
  height: 20%;
  top: 16%;
  left: 12%;

}

@media (max-width: 800px) {
  .SearchForm {
   top: 3%;
   left: 65%;
   z-index: 1;
  }
}

@media (max-width: 500px) {
  .SearchForm {
    top: 1.8%;
    width: 10%;
    left: 56%;
    z-index: 1;

    .SearchButton {
      position: fixed;
      top: .85%;
      right: 0;
    }
  }
}