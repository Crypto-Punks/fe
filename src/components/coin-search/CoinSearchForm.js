import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from './CoinSearchForm.css';


const CoinSearchForm = ({ handleSubmit }) => {
  const [searchTerm, setSearchTerm] = useState('');

  return (
    <form 
      className={styles.SearchForm} 
      onSubmit={event => {
        setSearchTerm('');
        handleSubmit(event, searchTerm);
      }}>
      <input type='text' value={searchTerm} onChange={({ target }) => setSearchTerm(target.value)} />
      <button>🔍</button>
    </form>
  );
};

CoinSearchForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

export default CoinSearchForm;
