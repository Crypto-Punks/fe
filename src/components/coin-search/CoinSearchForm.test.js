import React from 'react';
import { shallow } from 'enzyme';
import CoinSearchForm from './CoinSearchForm';

const data = {
  handleSubmit: () => {}
};

describe('CoinSearchForm component', () => {
  it('renders CoinSearchForm', () => {
    const wrapper = shallow(<CoinSearchForm { ...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
