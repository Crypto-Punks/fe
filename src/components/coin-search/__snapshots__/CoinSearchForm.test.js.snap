// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinSearchForm component renders CoinSearchForm 1`] = `
<form
  className="SearchForm"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="SearchButton"
  >
    🔍
  </button>
</form>
`;
